content/en/blog/2024/prom-and-otel/index.md

```diff
@@ -0,0 +1,670 @@
+ecosystem.
+
+Note that the PodMonitor and ServiceMonitor are not useful for cluster-wide
+metrics collection, such as for Kubelet metrics collection. In that case, you
+still have to rely on Prometheus scrape configs in the Collector’s
+[Prometheus Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/README.md).
+
+### Configuration
+
+The following is the YAML config for the OTel Collector CR. Note that this
+Collector is running in a namespace called `opentelemetry`, but it can run in
+whatever namespace you like.
+
+The main components are:
+
+- **mode:** This is one of four
+  [OTel Collector deployment modes supported by the Operator](https://github.com/open-telemetry/opentelemetry-operator?tab=readme-ov-file#deployment-modes):
+  Sidecar, Deployment, StatefulSet and DaemonSet.
+- **targetallocator:** This is where you configure the Target Allocator. Note
+  that the
+  [Target Allocator only works for the Deployment, DaemonSet, and StatefulSet modes](https://www.youtube.com/watch?v=Uwq4EPaMJFM).
+- **config:** This is where you configure the OTel Collector’s config YAML.
+
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: otelcol
+  namespace: opentelemetry
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+    serviceAccount: opentelemetry-targetallocator-sa
+    prometheusCR:
+      enabled: true
+  config: |
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+          http:
+      prometheus:
+        config:
+          scrape_configs:
+          - job_name: 'otel-collector'
+            scrape_interval: 30s
+            static_configs:
+            - targets: [ '0.0.0.0:8888' ]
+        target_allocator:
+          endpoint: http://otelcol-targetallocator
+          interval: 30s
+          collector_id: "${POD_NAME}"
+…
+```
+
+To use the Target Allocator, you need to set `spec.targetallocator.enabled` to
+`true`. (See previous note about supported modes.)
+
+Next, you need to make sure that the Prometheus receiver of the deployed
+Collector is made aware of the Target Allocator in the Collector config section
+of the spec by setting the `target_allocator.endpoint`:
+
+```yaml
+receivers:
+  prometheus:
+    config:
+      scrape_configs:
+        - job_name: 'otel-collector'
+          scrape_interval: 30s
+          static_configs:
+            - targets: ['0.0.0.0:8888']
+    target_allocator:
+      endpoint: http://otelcol-targetallocator
+      interval: 30s
+      collector_id: '${POD_NAME}'
+```
+
+The Target Allocator endpoint that the Prometheus receiver config is pointing to
+is a concatenation of the OTel Collector’s name (`otelcol`, in our case) and the
+`-targetallocator` suffix.
+[(This is now done automatically for users!)](https://github.com/open-telemetry/opentelemetry-operator/blob/main/pkg/featuregate/featuregate.go#L58-L65)
```

Comment:
What is the "this" that you are referring to that is "now done automatically"?
Also, is linking into a code file the best reference you have at hand?
I'm not sure I find the link destination that helpful to convince me/a reader of your statement. 
Also note that the lines that you refer to aren't in the code anymore 🤷🏼‍♂️ 
Maybe just drop the link, unless you can link to some docs or a blog post. WDYT?

Reply:
Yeah, good point. Let's drop it then.